src/core/ddsi/src/ddsi_typelib.c

```diff
@@ -244,10 +244,26 @@ static bool ti_to_pairs_equal (const dds_sequence_DDS_XTypes_TypeIdentifierTypeO
   if (a->_length != b->_length)
     return false;
 
-  struct dds_cdrstream_desc desc;
-  dds_cdrstream_desc_from_topic_desc (&desc, &DDS_XTypes_TypeObject_desc);
-
-  for (uint32_t n = 0; n < a->_length; n++)
+  // Can't do this (yet) is as a compile time constant because we need
+  // DDS_XTypes_TypeObject_desc.m_ops and the value is not visible here
+  const struct dds_cdrstream_desc desc = {
+    .size = sizeof (DDS_XTypes_TypeObject),
+    .align = dds_alignof (DDS_XTypes_TypeObject),
+    .flagset = DDS_TOPIC_CONTAINS_UNION,
+    .keys = {
+      .nkeys = 0,
+      .keys = NULL
+    },
+    .ops = {
+      .nops = 0, // serializer VM doesn't used it during execution
+      .ops = (uint32_t *) DDS_XTypes_TypeObject_desc.m_ops
+    },
+    .opt_size_xcdr1 = 0,
+    .opt_size_xcdr2 = 0
+  };
```

Comment:
We could add an option in IDLC to generate a `dds_cdrstream_desc` in addition to the topic descriptor type and include this in the generated meta-data for type info/object/map in DDSI. Then we can avoid this transformation, and also for a few other instances where a type object/information/map is serialized.